R/MzTabReader.R

```diff
@@ -106,8 +106,15 @@ getParameters = function()
   
   # copy the whole MTD for now, since R likes shallow copies and we are about to rename columns by reference (see ?setnames)
   res = data.table::copy(.self$sections[["MTD"]])
-  res = rbind(res, data.frame(key= "fasta file", value = paste(basename(unique(.self$sections$PSM$database)), collapse=";")))
-  res = res[-(grep("custom", res$key)),]
+  if (!is.na(unique(.self$sections$PSM$database))) {
+    res = rbind(res, data.frame(key= "fasta file", value = paste(basename(unique(.self$sections$PSM$database)), collapse=";")))
+  }
+  else {
+    res = rbind(res, data.frame(key= "fasta file", value = "NULL"))
+  }
+  if (any(grepl("custom", res$key))){
+    res = res[-(grep("custom", res$key)),] 
+  }
```

Comment:
you can make this a one-liner using
`res = res[grep("^custom", res$key, invert = TRUE),]`

Reply:
note the extra `^` just to make it more specific